/**
 * e8yes demo web server.
 *
 * <p>Copyright (C) 2020 Chifeng Wen {daviesx66@gmail.com}
 *
 * <p>This program is free software: you can redistribute it and/or modify it under the terms of the
 * GNU General Public License as published by the Free Software Foundation, either version 3 of the
 * License, or (at your option) any later version.
 *
 * <p>This program is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY;
 * without even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
 * GNU General Public License for more details.
 *
 * <p>You should have received a copy of the GNU General Public License along with this program. If
 * not, see <http://www.gnu.org/licenses/>.
 */

syntax = "proto3";

package e8;

import "chat_message.proto";
import "user_profile.proto";

message InvitationReceivedMessage {
    UserPublicProfile inviter = 1;
}

message InvitationAcceptedMessage {
    UserPublicProfile invitee = 1;
}

message UnreadChatMessage {
    repeated ChatMessageThread message_threads = 1;
}

message RealTimeMessageContent {
    oneof notification_content {
        InvitationReceivedMessage invitation_received = 1;
        InvitationAcceptedMessage invitation_accepted = 2;
        UnreadChatMessage unread_chat = 3;
    }
}

message RealTimeMessage {
    int64 real_time_message_id = 1;
    int64 created_at = 2;
    int64 target_user_id = 3;
    RealTimeMessageContent content = 4;
}
